source/Halibut.Tests/Support/TestCases/ClientAndServiceTestCasesBuilder.cs

```diff
@@ -35,7 +39,24 @@ public IEnumerable<ClientAndServiceTestCase> Build()
                             recommendedIterations = StandardIterationCount.ForServiceType(serviceConnectionType, clientServiceTestVersion);
                         }
 
-                        yield return new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion);
+                        if (!forceProxyClientTypes.Any())
+                        {
+                            yield return new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion, null);
+                        }
+                        else
+                        {
+                            if (clientServiceTestVersion.IsPreviousClient())
+                            {
+                                yield return new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion, null);
```

Comment:
Do we end up with 2 tests that test the same thing for previous client in this case?

Reply:
I think we only yield twice in the next foreach, here we yeild once with null